eth/filters/api.go

```diff
@@ -132,9 +133,10 @@ func (api *FilterAPI) NewPendingTransactionFilter() rpc.ID {
 	return pendingTxSub.ID
 }
 
-// NewPendingTransactions creates a subscription that is triggered each time a transaction
-// enters the transaction pool and was signed from one of the transactions this nodes manages.
-func (api *FilterAPI) NewPendingTransactions(ctx context.Context) (*rpc.Subscription, error) {
+// NewPendingTransactions creates a subscription that is triggered each time a
+// transaction enters the transaction pool. If fullTx is true the full tx is
+// sent to the client, otherwise the hash is sent.
+func (api *FilterAPI) NewPendingTransactions(ctx context.Context, fullTx bool) (*rpc.Subscription, error) {
```

Comment:
This is a breaking change. I suggest adding a new methods to get the full tx object.

Reply:
Yes, any suggestions? An alternative could be to interpreted the missing `fullTx` arg as `fullTx=false`. This would make the change non-breaking, but I don't think this is possible with the `rpc` package rn.

```diff
// SubscribePendingTransactionHashes subscribes to new pending transaction hashes.
func (ec *Client) SubscribePendingTransactionHashes(ctx context.Context, ch chan<- common.Hash) (*rpc.ClientSubscription, error) {
-	return ec.c.EthSubscribe(ctx, ch, "newPendingTransactions", false)
+	return ec.c.EthSubscribe(ctx, ch, "newPendingTransactions")
}
```

Reply:
Seems like optional args are possible ([relevant code](https://github.com/ethereum/go-ethereum/blob/master/rpc/json.go#L315)) by making the fullTx arg a pointer. This way `newPendingTransactions` would keep its current behavior and would be an alias for `newPendingTransactions, false`. @s1na Is that okay?

Reply:
Ah yes that sounds good to me